cpp/include/raft/spectral/kmeans.hpp

```diff
@@ -0,0 +1,898 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
```

Comment:
It would be nice to see this in a more shared location, outside of spectral. Maybe `raft/cluster/kmeans.hpp`? We can keep `cluster_solver_t` in spectral, but I think it'll also make it easier to integrate cuml's kmeans once it's moved over.

Reply:
But what if both must coexist? Then we'd have a `cugraph`-specific `kmeans` and a `cuml` one. Which should be kept separately, no? 

In any case, I can make the move once it's been clarified where. Please specify the desired location to create and move to. `raft/cluster/kmeans.hpp` confirmed?

Reply:
I'm okay keeping this one where it is until we figure out what path we plan to take with kmeans.